### PR TITLE
Update PR template wording about changelog

### PR DESCRIPTION
**Pull request by cijothomas**
A lot of times, PR desc. is not updated after PR itself is changed based on feedbacks, etc.
Changelog.md is not clear at times - we need to encourage it to be self-explanatory wherever possible, or link to PR description where the details are described.

### CODE REVIEW

File: src/OpenTelemetry/Metrics/Tags.cs [hunk 1]
```diff
@@ -52,6 +52,8 @@ public Tags(KeyValuePair<string, object?>[] keyValuePairs)
 
     public readonly bool Equals(Tags other)
     {
+        // Equality check is done without sorting, so the order of tags matters.
```
Comment (cijothomas) [inline src/OpenTelemetry/Metrics/Tags.cs:55]: not sure if others appreciate such code comments! Happy to remove if not required.

File: src/OpenTelemetry/Metrics/AggregatorStore.cs [hunk 2]
```diff
@@ -558,8 +558,11 @@ private int LookupAggregatorStoreForDeltaWithReclaim(KeyValuePair<string, object
         {
             if (length > 1)
             {
-                // Note: We are using storage from ThreadStatic, so need to make a deep copy for Dictionary storage.
-                // Create or obtain new arrays to temporarily hold the sorted tag Keys and Values
+                // Create or obtain new arrays to temporarily hold the sorted tag Keys and Values.
```
Comment (cijothomas) [inline src/OpenTelemetry/Metrics/AggregatorStore.cs:561]: this fixed an incorrect comment (there was no deep copy here.), and also expanded further on "why"... Still unsure, if folks find such code comments useful or distracting. Let me know.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,13 +1,23 @@
...
+  [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md)
+  guidelines followed (license requirements, nullable enabled, static analysis,
+  etc.)
 * [ ] Unit tests added/updated
-* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes
+* [ ] For all non-trivial modifications, please ensure the `CHANGELOG.md` file is
+  duly updated. Each changelog entry should be self-explanatory, providing a
+  concise, yet comprehensive overview of the changes made. In instances where a
+  change is too extensive for a brief changelog description, detail the
+  modifications within the pull request description and add link to it.
```
Comment (CodeBlanch) [inline .github/PULL_REQUEST_TEMPLATE.md:22]: I get what you are going for, but I feel like this is a lot of text to have in the template. Could we link to a section in contributing guide which has more details for users who need help?

File: src/OpenTelemetry/Metrics/AggregatorStore.cs (hunk 2 above)
Comment (CodeBlanch) [reply src/OpenTelemetry/Metrics/AggregatorStore.cs:561]: Did you mean to include these on this PR? Feel like they should be their own thing.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (cijothomas) [reply .github/PULL_REQUEST_TEMPLATE.md:22]: Thats what I was thinking initially... But then thought that not many people read the contributing guide, but PR template is harder to ignore....

Comment (Kielek) [reply .github/PULL_REQUEST_TEMPLATE.md:22]: IMO link(s) to PR should be always added

Comment (cijothomas) [reply .github/PULL_REQUEST_TEMPLATE.md:22]: Agree to "Link to PR is always needed". Will modify wording.

Comment (CodeBlanch) [reply .github/PULL_REQUEST_TEMPLATE.md:22]: > Thats what I was thinking initially... But then thought that not many people read the contributing guide, but PR template is harder to ignore....

IMO it is too spammy and it is going to be annoying to the regular contributors and people reviewing PRs. It should be short here. If people don't follow it, that is fine. Part of the approver/maintainer workflow when reviewing PRs is to verify it is all followed. Having it in contributing would help those who read it, but also gives us something we can link people when they don't follow it (or do it wrong) 🤣 

My suggestion is to do like...

```md
* [ ] [Appropriate CHANGELOG files updated for non-trivial changes](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md#changelog-modification-guidelines)
```

Not a blocker for me, but no guarantee I won't try to change it in the future if it becomes annoying 😄

Comment (cijothomas) [reply .github/PULL_REQUEST_TEMPLATE.md:22]: no, I think this is more apt. Let me update accordingly.

**Comment by cijothomas** [comment]
Changed to draft, as I need to spend some time to implement the suggestions and come back.

**Pull request closed without merging**